src/chart/generateCategoricalChart.tsx

```diff
@@ -1659,10 +1645,8 @@ export const generateCategoricalChart = ({
     };
 
     triggerSyncEvent(data: CategoricalChartState) {
-      const { syncId } = this.props;
-
-      if (!_.isNil(syncId)) {
-        eventCenter.emit(SYNC_EVENT, syncId, this.uniqueChartId, data);
+      if (this.props.syncId !== undefined) {
```

Comment:
```suggestion
      if (this.props.syncId != undefined) {
```
I assume we would also want to catch null, right?

Reply:
No, there shouldn't be any null. Typing errors should be caught by TypeScript.